### PR TITLE
Conda build for head/headless x with bullet/without bullet, CI test, issues fixed

### PR DESCRIPTION
**Pull request by mathfac**
#  Motivation and Context
- Made Linux binaries build for head / headless x with bullet / without bullet
- Improved Conda build readme
- Use Frozen `conda-build` lib version that works with current setup
- Added conda binaries build step to CI without upload to Anaconda upload for testing purposes. 
- Next Conda workflows are supported now:
Screenshot 2020-07-01 22 57 29

## How Has This Been Tested
- Built v0.1.5 binaries.
- Added CI build step

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: conda-build/README.md [hunk 1]
```diff
@@ -9,19 +10,19 @@ To then download the package, run ```conda install -c aihabitat -c conda-forge h
 
 ### Building for Linux
 
-The process is almost the same for linux; there is a corresponding python script for starting things off, however we use a docker container to do the builds. There is a dockerfile in this directory that you can use to create a container.
+The process is almost the same for linux; there is a corresponding python script for starting things off. You can use Ubuntu 16.04 or  a docker container to do the builds. There is a dockerfile in this directory that you can use to create a container.
```
Comment (erikwijmans) [inline conda-build/README.md:13]: This is likely untrue unfortunately.  Using that old version of linux in the docker container is important to make the binaries be able to run on different platforms.  Maybe 16.04 is old enough to be the base these days, but we would need to do a lot of verification.

Comment (mathfac) [reply conda-build/README.md:13]: Last binaries I built for v0.1.5 release were based on Ubuntu 16.04 that is used by our CI image.

**Comment by mathfac** [comment]
I added the CI step that build conda package (headless + with bullet) for each PR, it takes 10 minutes and validates that all updates to `meta.xml` are there and our nightly build is running ok:
Screenshot 2020-06-29 10 34 34

The CI jobs that builds all 4 binaries (head/headless x with bullet/without bullet) for Linux added to master only nightly runs (I will add filters) and finished successfully: 
Screenshot 2020-06-29 10 34 44

File: conda-build/README.md (hunk 1 above)
Comment (Skylion007) [reply conda-build/README.md:13]: @mathfac That will work for most modern Linux releases, but may not work for certain older distributions. Particularly those that are stable for very long periods of time (CentOS for example). Older versions of Debian and Ubuntu may also not work.

Still probably best to build the official releases with that Docker for that reason.

**Comment by mathfac** [comment]
Made Linux binaries build for head / headless x with bullet / without bullet
Improved Conda build readme
Use Frozen conda-build lib version that works with current setup
Added conda binaries build step to CI without upload to Anaconda upload for testing purposes.

**Pull request merged**